File: javascript/test/Question2.test.js
const chai = require('chai');
const assert = chai.assert;
const utopianTree = require('../question2')

describe('Question 2', () => {
  it ('Can get the tree height for test case 1', done => {
    let result = utopianTree(0)
    assert.equal(result, 1)
    done()
  })

  it ('Can get the tree height for test case 2', done => {
    let result = utopianTree(1)
    assert.equal(result, 2)
    done()
  })

  it ('Can get the tree height for test case 3', done => {
    let result = utopianTree(4)
    assert.equal(result, 7)
    done()
  })
})

File: injava/test/Question2Test.java
package injava.test;

import static org.junit.Assert.*;
import org.junit.Before;
import org.junit.Test;
import injava.Question2;

public class Question2Test {
    
    private Question2 question;

    @Before
    public void before() {
        this.question = new Question2();
    }
    
    @Test
    public void test1 () {
        int result = this.question.utopianTree(0);
        assertEquals(1, result);
    }

    @Test
    public void test2 () {
        int result = this.question.utopianTree(1);
        assertEquals(2, result);
    }

    @Test
    public void test3 () {
        int result = this.question.utopianTree(4);
        assertEquals(7, result);
    }
}